### DISCUSSION TITLE
Error in devtools::install()

### DISCUSSION THREAD

**Discussion by 2199105T** [IBAHCM/RPiR#320] [Archive] [answered]
Hi @richardreeve @soniamitchell,

My code was running fine then all of a sudden I can't run devtools::install() without this error message:

```{r}
Error: unable to load R code in package ‘sdtonnarDiversity’
Execution halted
ERROR: lazy loading failed for package ‘sdtonnarDiversity’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/sdtonnarDiversity’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/sdtonnarDiversity’
Error in (function (command = NULL, args = character(), error_on_status = TRUE,  : 
  System command 'R' failed, exit status: 1, stdout & stderr were printed
```

I'm not sure what's going wrong! It was working fine not too long ago!

**Comment by soniamitchell** [comment, marked as answer]
Did you push everything to GitHub? or is there an R file that's not a function inside your R directory? This is all I can find...

https://stackoverflow.com/questions/32122780/lazy-loading-error-in-r-package

If that doesn't work, I suggest restarting R 👍

**Reply by 2199105T** [reply]
Hi @soniamitchell,

I have pushed it now but it still doesn't seem to be working.

The only files in my R directory are my richness and  Simpson functions and my package documentation.

**Reply by soniamitchell** [reply]
Hi @2199105T, line 19 of `p1_species_richness.R`, the `test_function` argument should not be a string. Actually `assert_no_globals` should either give a more informative error or `?assert_no_globals` should be better documented 👍

**Reply by 2199105T** [reply]
Hi @soniamitchell 

I've managed to get it working for now! Are we also okay to use the vegan package to calculate Simpson's or do we need to calculate it using the equation etc?

**Reply by soniamitchell** [reply]
Hey Sarah, you should be using the equation and building your own functions 👍
